src/components/uploadpost/UploadPost.jsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import { toast, ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { useNavigate } from 'react-router-dom';
+
+import {
+  StyleFormConatiner,
+  StyleInput,
+  CancelButton,
+  DoneButton,
+  LargeInput,
+  StyledLabel,
+  InputRow
+} from './StyledUpload.js';
+import { supabase } from '../../../supabase/supabase.js'; // supabase 사용하려면
+import SelectCategory from './../SelectCategory/SelectCategory';
+import { useDispatch, useSelector } from 'react-redux';
+
+const UploadPost = () => {
+  const [title, setTitle] = useState('');
+  const [nickname, setNickname] = useState('');
+  const [passwords, setPassword] = useState('');
+  const [content, setContent] = useState('');
+  const [time, setTime] = useState('');
+  const [participants, setParticipants] = useState('');
+  const [coverImage, setCoverImage] = useState('');
+  const [imageUrl, setImageUrl] = useState('');
+
+  const category = useSelector((state) => state.UploadPostSlice.category);
+  const lng = useSelector((state) => state.mapSlice.lng);
+  const lat = useSelector((state) => state.mapSlice.lat);
+
+  const navigate = useNavigate();
+
+  const handleClickConfirm = async () => {
+    let imageUploadUrl = null;
+
+    if (coverImage) {
+      imageUploadUrl = await uploadImage(coverImage);
+      console.log('Uploaded image URL:', imageUploadUrl);
+    }
+
+    const newObject = {
+      time: time,
+      title: title,
+      head_count: participants,
+      body: content,
+      user_id: nickname,
+      user_pw: passwords,
+      category,
+      map_lng: lng,
+      map_lat: lat,
+      image: imageUploadUrl // 이미지 URL을 설정합니다.
+    };
+
+    console.log(newObject);
+    await writePost(newObject);
+  };
+
+  const uploadImage = async (file) => {
+    const fileName = `${Date.now()}`;
+    const filePath = `${fileName}`;
```

Comment:
```suggestion
    const filePath = `${Date.now()}`;
```
두 줄을 합칠 수 있을 것 같습니다!
이렇게 한다면 메모리 공간 한개를 아낄 수 있습니다!